tensorflow/compiler/xla/service/hlo_evaluator_test.cc

```diff
@@ -2567,7 +2591,90 @@ ENTRY main {
   Literal arg = LiteralUtil::CreateR1<bfloat16>(
       {bfloat16(1.0f), bfloat16(3.0f), bfloat16(-2.0f), bfloat16(42.0f)});
   Literal expected = LiteralUtil::CreateR0<bfloat16>(bfloat16(44.0f));
-  EXPECT_TRUE(LiteralTestUtil::Equal(expected, Evaluate({&arg})));
+  TF_ASSERT_OK_AND_ASSIGN(Literal result, Evaluate({&arg}));
+  EXPECT_TRUE(LiteralTestUtil::Equal(expected, result));
+}
+
+TEST_P(HloEvaluatorTest, DontFailOnCall) {
+  // The following test used to trigger unimplemented error instead of returning
+  // false. This was due to HandleCall in HloEvaluator using ConsumeValueOrDie
+  // to get a literal value, however this is not possible with RNG.
+  const string hlo_text = R"(
```

Comment:
This comment is not describing the current state and could be confusing. Instead I wonder if just saying that "RNG triggers unimplemented error within the Call, and we verify that the Evaluator does fail in such case" would be better?

Same for the comments for the test cases below as well.